posthog/temporal/tests/batch_exports/test_http_batch_export_workflow.py

```diff
@@ -0,0 +1,466 @@
+            ):
+                m.post(TEST_URL, status=200, callback=mock_server.post, repeat=True)
+
+                await activity_environment.client.execute_workflow(
+                    HttpBatchExportWorkflow.run,
+                    inputs,
+                    id=workflow_id,
+                    task_queue=settings.TEMPORAL_TASK_QUEUE,
+                    retry_policy=RetryPolicy(maximum_attempts=1),
+                    execution_timeout=dt.timedelta(seconds=10),
+                )
+
+    runs = await afetch_batch_export_runs(batch_export_id=http_batch_export.id)
+    assert len(runs) == 1
+
+    run = runs[0]
+    assert run.status == "Completed"
+
+    await assert_clickhouse_records_in_mock_server(
+        mock_server=mock_server,
+        clickhouse_client=clickhouse_client,
+        team_id=ateam.pk,
+        data_interval_start=data_interval_start,
+        data_interval_end=data_interval_end,
+        exclude_events=exclude_events,
+    )
+
+
+async def test_http_export_workflow_handles_insert_activity_errors(ateam, http_batch_export, interval):
+    """Test that HTTP Export Workflow can gracefully handle errors when POSTing to HTTP Endpoint."""
+    data_interval_end = dt.datetime.fromisoformat("2023-04-25T14:30:00.000000+00:00")
+
+    workflow_id = str(uuid4())
+    inputs = HttpBatchExportInputs(
+        team_id=ateam.pk,
+        batch_export_id=str(http_batch_export.id),
+        data_interval_end=data_interval_end.isoformat(),
+        interval=interval,
+        **http_batch_export.destination.config,
+    )
+
+    @activity.defn(name="insert_into_http_activity")
+    async def insert_into_http_activity_mocked(_: HttpInsertInputs) -> str:
+        raise ValueError("A useful error message")
+
+    async with await WorkflowEnvironment.start_time_skipping() as activity_environment:
+        async with Worker(
+            activity_environment.client,
+            task_queue=settings.TEMPORAL_TASK_QUEUE,
+            workflows=[HttpBatchExportWorkflow],
+            activities=[
+                create_export_run,
+                insert_into_http_activity_mocked,
+                update_export_run_status,
+            ],
+            workflow_runner=UnsandboxedWorkflowRunner(),
+        ):
+            with pytest.raises(WorkflowFailureError):
+                await activity_environment.client.execute_workflow(
+                    HttpBatchExportWorkflow.run,
+                    inputs,
+                    id=workflow_id,
+                    task_queue=settings.TEMPORAL_TASK_QUEUE,
+                    retry_policy=RetryPolicy(maximum_attempts=1),
+                )
+
+        runs = await afetch_batch_export_runs(batch_export_id=http_batch_export.id)
+        assert len(runs) == 1
+
+        run = runs[0]
+        assert run.status == "Failed"
+        assert run.latest_error == "ValueError: A useful error message"
+
+
+async def test_http_export_workflow_handles_cancellation(ateam, http_batch_export, interval):
```

Comment:
This test is cheating a bit: Cancellations are communicated to the worker on `hearbeat` callsl, but we are not calling `heartbeat` yet.

The name of the test kind of implies that we support cancellation when we actually don't.

Reply:
Still, useful to know we set the appropriate status and error.

Reply:
> The name of the test kind of implies that we support cancellation when we actually don't.

Ah interesting, FWIW I took this from Postgres tests, if it matters there too.